include/z64cutscene.h

```diff
@@ -344,6 +344,42 @@ typedef enum CutsceneDestination {
     /* 0x77 */ CS_DEST_ZELDAS_COURTYARD_RECEIVE_LETTER
 } CutsceneDestination;
 
+/*
+correlation with `enum SceneLayer`:
+  CS_INDEX_MANUAL:     [SCENE_LAYER_CHILD_DAY .. SCENE_LAYER_ADULT_NIGHT]
+  CS_INDEX_SCRIPTED_*: SCENE_LAYER_CUTSCENE_FIRST
+
+correlation with `enum EntranceIndex`
+  CS_INDEX_MANUAL:     base entrance index + scene layer
+  CS_INDEX_SCRIPTED_*: base entrance index + SCENE_LAYER_CUTSCENE_FIRST + (cutscene index & 0xF)
+
+`z_select.c` naming:
+  CS_INDEX_NONE:       "Stage: night"
+  CS_INDEX_LAST:       "Stage: day"
+  CS_INDEX_SCRIPTED_*: "Stage demo 0*", between [0xFFF0 .. 0xFFFA]
+*/
+typedef enum CutsceneIndex {
+    CS_INDEX_NONE = 0x0000,
+    CS_INDEX_MANUAL = 0xFFEF,
```

Comment:
current naming is an observation:
```
// z_demo.c:180
// i.e. CS_INDEX_MANUAL or lower
void Cutscene_UpdateManual(...)
    if (gSaveContext.save.cutsceneIndex < CS_INDEX_SCRIPTED_0)
        sManualCutsceneHandlers[csCtx->state](...);

// z_demo.c:215
// i.e. CS_INDEX_SCRIPTED_0 or higher
void Cutscene_UpdateScripted(...)
    if (gSaveContext.save.cutsceneIndex >= CS_INDEX_SCRIPTED_0)
        sScriptedCutsceneHandlers[csCtx->state](...);
```

Reply:
cutsceneIndex never gets assigned to 0xFFEF though. It is only ever assigned to nextCutsceneIndex, and then the value is only used once in Play_Init where if nextCutsceneIndex is 0xFFEF then it's value is not copied over.

Reply:
ok, thanks. you are right here, i'm conflating these two together, because they are somewhat related
```
// `z_play.c:328`
if (gSaveContext.nextCutsceneIndex != CS_INDEX_MANUAL) {
    gSaveContext.save.cutsceneIndex = gSaveContext.nextCutsceneIndex;
    gSaveContext.nextCutsceneIndex = CS_INDEX_MANUAL;
}
```
i'll look closer

Reply:
https://github.com/zeldaret/oot/pull/2286#discussion_r1861134687

it rather means *"there's nothing scheduled **next**, feel free to employ it"*
- `z_play.c` passes it's value to `cutsceneIndex` if it's not "empty"
  https://github.com/zeldaret/oot/blob/279759f4815c828a1bb283c93ee62ad69fc6750b/src/code/z_play.c#L327-L329
- `z_door_warp1.c` "fills" `nextCutsceneIndex` if it is `empty`
  https://github.com/zeldaret/oot/blob/279759f4815c828a1bb283c93ee62ad69fc6750b/src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c#L492
- `z_common_data.c` and `z_file_choose.c` just clear `nextCutsceneIndex`
- at the same time nothing in the code uses it in the sense of *"it's marked `_NO_NEXT`; please, do not schedule anything"*

Reply:
> at the same time nothing in the code uses it in the sense of "it's marked _NO_NEXT; please, do not schedule anything"

Isn't this exactly what the first code block you posted is doing? The "scheduled" value of `nextCutsceneIndex` is only copied to `cutsceneIndex` if it's not 0xFFEF. (Although I intended the name to be read as "no next cutscene" rather than "no next")

Reply:
that's why i said that: it sounds to me like *"no next cutscene, please"*, rather than *"there's no next cutscene"*; at least ambigous

at the same time `cutscene != empty` reads, again in my head, as *"if scheduled, consume the queued `CS_INDEX`"*

Reply:
that's why i said that: it sounds to me like *"no next cutscene, please"*, rather than *"there's no next cutscene"*; at least ambigous

at the same time `cutscene != empty` reads, again in my head, as *"if scheduled, consume the queued `CS_INDEX`"*

also, from your proposal https://github.com/zeldaret/oot/pull/2286#discussion_r1863815412
> // Sentinel value used for `nextCutsceneIndex` to indicate that the `cutsceneIndex` should not be changed on scene load. Otherwise, `nextCutsceneIndex` will be copied to `cutsceneIndex`.
#define CS_INDEX_NO_NEXT_CUTSCENE 0xFFEF

to me it shouldn't say "indicate that cutscene should not be changed", but rather "indicate that there's nothing to be changed to". a subtle difference, less imperative phrasing, simply signifying the absence of data rather than a command to stop doing somthing